internal/govulncheckshim/client.go

```diff
@@ -0,0 +1,110 @@
+package govulncheckshim
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/google/osv-scanner/pkg/models"
+	"golang.org/x/vuln/client"
+	gvcOSV "golang.org/x/vuln/osv"
+)
+
+type localSource struct {
+	vulnList         []models.Vulnerability
+	vulnsByID        map[string]*models.Vulnerability
+	vulnsByAlias     map[string][]*models.Vulnerability
+	vulnsByModule    map[string][]*models.Vulnerability
+	lastModifiedTime time.Time
+	client.Client
+}
+
+func newClient(vulns []models.Vulnerability) *localSource {
```

Comment:
I guess the precondition here is that all vulns apply to the program under consideration?

I mention this for the following reason.  A single `gvcOSV.Entry` can contain information on vulnerabilities in 1) [different modules/packages](https://github.com/golang/vulndb/blob/cf9ad2b3dc5b7ffca7e2fcff0de336bcd1e25e62/data/reports/GO-2020-0036.yaml) and/or 2) even different ranges within the same package. That is why each `gvcOSV.Affected` defines module, package, range, symbols, and platform info. govulncheck hence does a [fine grained filtering](https://github.com/golang/vuln/blob/8fb696828224a4da6b1492cd467a841f7230f0b5/vulncheck/vulncheck.go#L313) based on module info in the user program.

It could be that this is taken care of somewhere higher in the stack or that it does not matter for your current usage. I wanted to mention it as it is quite subtle.

Reply:
`vulns` should contain all the osv entries that affects any packages in the go.mod file of the program.

Since it's the OSV entry, it should contain the almost identical information to gvcOSV.Entry (e.g. GetByID/Alias/Module will return GO-2020-0036.yaml in it's entirety if it matches).